woodoku/tests/entity/test_score_agent.py

```diff
@@ -0,0 +1,223 @@
+from typing import Iterable, Tuple
+
+import pytest
+from woodoku.entity.score_agent import GROUP_POINTS, STREAK_POINTS, ScoreAgent
+
+# Scoring Rules:
+#   1. 1 point for each block placed
+#   2. A single groups scores 18 points
+#   3. Each streak adds 10 points on top of the previous streak
+#   4. Each additional combo adds 28 points
+
+
+# pytest setup
+# tuples are (No. blocks placed, No. groups completed)
+@pytest.fixture(name="scorekeeper")
+def fixture_scorekeeper() -> ScoreAgent:
+    return ScoreAgent()
+
+
+# Question... (just wanna know if there is a better way)
+# Is there any way to use Mock or MagicMock to set the private state instead of
+# a hack like this or accessing the private variable
+# for example: I want to set up so that before testing the scorekeeper has a
+# score of 42 and current streak of 1
+@pytest.fixture(name="setup_and_apply_winnings", autouse=True)
```

Comment:
I did find a way to mock property [in MagicMock](https://stackoverflow.com/questions/55512708/how-to-properly-mock-private-members-of-a-class) but I think mocking is not applicable in testing scoreAgent since once we make a scoreAgent a mock object, we can no longer call calculate_winning which is the function to test.